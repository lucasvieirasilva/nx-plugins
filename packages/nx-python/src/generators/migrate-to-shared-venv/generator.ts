import {
  readJsonFile,
  formatFiles,
  generateFiles,
  getProjects,
  Tree,
  ProjectConfiguration,
} from '@nx/devkit';
import path from 'path';
import { Schema } from './schema';
import { parse, stringify } from '@iarna/toml';
import chalk from 'chalk';
import { PoetryPyprojectToml } from '../../provider/poetry';
import { UVPyprojectToml } from '../../provider/uv/types';
import { getProvider } from '../../provider';
import { IProvider } from '../../provider/base';

async function addFiles(host: Tree, options: Schema) {
  const packageJson = await readJsonFile('package.json');

  const templateOptions = {
    ...options,
    template: '',
    dot: '.',
    packageName: packageJson.name,
    description: packageJson.description || '',
  };

  generateFiles(
    host,
    path.join(__dirname, 'files', 'base'),
    '.',
    templateOptions,
  );

  if (options.packageManager === 'poetry') {
    generateFiles(
      host,
      path.join(__dirname, 'files', 'poetry'),
      '.',
      templateOptions,
    );
  } else if (options.packageManager === 'uv') {
    generateFiles(
      host,
      path.join(__dirname, 'files', 'uv'),
      '.',
      templateOptions,
    );
  }
}

type LockUpdateTask = () => Promise<void>;

function updatePoetryPyprojectRoot(
  host: Tree,
  options: Schema,
  provider: IProvider,
): LockUpdateTask[] {
  const postGeneratorTasks = [];

  const rootPyprojectToml = parse(
    host.read('pyproject.toml').toString(),
  ) as PoetryPyprojectToml;

  for (const project of getProjects(host)) {
    const [, projectConfig] = project;
    const pyprojectTomlPath = path.join(projectConfig.root, 'pyproject.toml');
    if (host.exists(pyprojectTomlPath)) {
      const pyprojectToml = parse(
        host.read(pyprojectTomlPath).toString(),
      ) as PoetryPyprojectToml;

      rootPyprojectToml.tool.poetry.dependencies[
        pyprojectToml.tool.poetry.name
      ] = {
        path: projectConfig.root,
        develop: true,
      };
      if (options.moveDevDependencies) {
        postGeneratorTasks.push(
          movePoetryDevDependencies(
            pyprojectToml,
            rootPyprojectToml,
            host,
            pyprojectTomlPath,
            projectConfig,
            provider,
          ),
        );
      }
    }
  }

  host.write('pyproject.toml', stringify(rootPyprojectToml));

  return postGeneratorTasks;
}

function movePoetryDevDependencies(
  pyprojectToml: PoetryPyprojectToml,
  rootPyprojectToml: PoetryPyprojectToml,
  host: Tree,
  pyprojectTomlPath: string,
  projectConfig: ProjectConfiguration,
  provider: IProvider,
) {
  const devDependencies =
    pyprojectToml.tool.poetry.group?.dev?.dependencies || {};

  for (const devDependency of Object.keys(devDependencies)) {
    rootPyprojectToml.tool.poetry.group =
      rootPyprojectToml.tool.poetry.group || {};
    rootPyprojectToml.tool.poetry.group.dev = rootPyprojectToml.tool.poetry
      .group.dev || { dependencies: {} };
    rootPyprojectToml.tool.poetry.group.dev.dependencies[devDependency] =
      devDependencies[devDependency];
  }

  if (Object.keys(devDependencies).length > 0) {
    delete pyprojectToml.tool.poetry.group.dev;
  }
  host.write(pyprojectTomlPath, stringify(pyprojectToml));

  return async () => {
    console.log(
      chalk`  Updating ${pyprojectToml.tool.poetry.name} {bgBlue poetry.lock}...`,
    );
    await provider.lock(projectConfig.root);
    console.log(chalk`\n  {bgBlue poetry.lock} updated.\n`);
  };
}

function updateUvPyprojectRoot(host: Tree, options: Schema): LockUpdateTask[] {
  const postGeneratorTasks = [];

  const rootPyprojectToml = parse(
    host.read('pyproject.toml').toString(),
  ) as UVPyprojectToml;

  for (const project of getProjects(host)) {
    const [, projectConfig] = project;
    const pyprojectTomlPath = path.join(projectConfig.root, 'pyproject.toml');
    if (host.exists(pyprojectTomlPath)) {
      const pyprojectToml = parse(
        host.read(pyprojectTomlPath).toString(),
      ) as UVPyprojectToml;

      rootPyprojectToml.project.dependencies.push(pyprojectToml.project.name);
      rootPyprojectToml.tool ??= {};
      rootPyprojectToml.tool.uv ??= {};
      rootPyprojectToml.tool.uv.sources ??= {};
      rootPyprojectToml.tool.uv.sources[pyprojectToml.project.name] = {
        workspace: true,
      };
      rootPyprojectToml.tool.uv.workspace ??= {
        members: [],
      };
      rootPyprojectToml.tool.uv.workspace.members.push(projectConfig.root);

      for (const source of Object.keys(pyprojectToml.tool?.uv?.sources ?? {})) {
        if (pyprojectToml.tool.uv.sources[source].path) {
          pyprojectToml.tool.uv.sources[source] = { workspace: true };
        }
      }

      if (options.moveDevDependencies) {
        moveUvDevDependencies(pyprojectToml, rootPyprojectToml);
      }

      host.write(pyprojectTomlPath, stringify(pyprojectToml));
      host.delete(path.join(projectConfig.root, 'uv.lock'));
      host.delete(path.join(projectConfig.root, '.venv'));
    }
  }

  host.write('pyproject.toml', stringify(rootPyprojectToml));

  return postGeneratorTasks;
}

function moveUvDevDependencies(
  pyprojectToml: UVPyprojectToml,
  rootPyprojectToml: UVPyprojectToml,
) {
  const devDependencies = pyprojectToml?.['dependency-groups']?.dev || [];
  rootPyprojectToml['dependency-groups'] ??= {};
  rootPyprojectToml['dependency-groups'].dev ??= [];

  for (const devDependency of devDependencies) {
    if (
      rootPyprojectToml['dependency-groups'].dev.some(
        (dep) =>
          /^[a-zA-Z0-9-]+/.exec(dep)?.[0] ===
          /^[a-zA-Z0-9-]+/.exec(devDependency)?.[0],
      )
    ) {
      continue;
    }
    rootPyprojectToml['dependency-groups'].dev.push(devDependency);
  }

  if (pyprojectToml['dependency-groups']?.dev?.length) {
    delete pyprojectToml['dependency-groups'].dev;
  }

  if (Object.keys(pyprojectToml['dependency-groups'] || {}).length === 0) {
    delete pyprojectToml['dependency-groups'];
  }
}

async function generator(host: Tree, options: Schema) {
  const provider = await getProvider(
    host.root,
    undefined,
    host,
    undefined,
    options,
  );

  await provider.checkPrerequisites();

  await addFiles(host, options);
  const lockUpdateTasks: LockUpdateTask[] = [];
  if (options.packageManager === 'poetry') {
    lockUpdateTasks.push(...updatePoetryPyprojectRoot(host, options, provider));
  } else if (options.packageManager === 'uv') {
    lockUpdateTasks.push(...updateUvPyprojectRoot(host, options));
  }

  await formatFiles(host);

  return async () => {
    for (const task of lockUpdateTasks) {
      await task();
    }

    await provider.install();
  };
}

export default generator;
