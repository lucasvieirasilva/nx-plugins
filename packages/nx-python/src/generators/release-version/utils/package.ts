import { joinPathFragments } from '@nx/devkit';
import { IProvider } from '../../../provider/base';

export class Package {
  name: string;
  version: string;
  location: string;

  constructor(
    private readonly provider: IProvider,
    workspaceRoot: string,
    private workspaceRelativeLocation: string,
  ) {
    const metadata = provider.getMetadata(workspaceRelativeLocation);
    this.name = metadata.name;
    this.version = metadata.version;
    this.location = joinPathFragments(workspaceRoot, workspaceRelativeLocation);
  }

  getLocalDependency(depName: string): {
    collection: 'dependencies' | 'devDependencies' | 'optionalDependencies';
    groupKey?: string;
    spec: string;
  } | null {
    const depMatadata = this.provider.getDependencyMetadata(
      this.workspaceRelativeLocation,
      depName,
    );

    if (!depMatadata) {
      return null;
    }

    return {
      collection:
        depMatadata.group === 'main'
          ? 'dependencies'
          : depMatadata.group === 'dev'
            ? 'devDependencies'
            : 'optionalDependencies',
      spec: depMatadata.version,
    };
  }
}
