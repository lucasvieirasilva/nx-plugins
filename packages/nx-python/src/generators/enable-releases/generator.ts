import { getProjects, Tree, updateProjectConfiguration } from '@nx/devkit';
import path from 'path';
import { updateNxReleaseConfig } from '../utils';
import { Schema } from './schema';

async function generator(host: Tree, options: Schema) {
  for (const project of getProjects(host)) {
    const [projectName, projectConfig] = project;
    const pyprojectTomlPath = path.join(projectConfig.root, 'pyproject.toml');
    if (host.exists(pyprojectTomlPath)) {
      updateNxReleaseConfig(options, projectConfig);
      updateProjectConfiguration(host, projectName, projectConfig);
    }
  }
}

export default generator;
