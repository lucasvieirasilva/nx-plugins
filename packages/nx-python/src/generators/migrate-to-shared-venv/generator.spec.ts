import { vi, MockInstance } from 'vitest';
import '../../utils/mocks/cross-spawn.mock';
import * as poetryUtils from '../../provider/poetry/utils';
import * as uvUtils from '../../provider/uv/utils';
import { Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import generator from './generator';
import poetryProjectGenerator from '../project/generator';
import uvProjectGenerator from '../uv-project/generator';
import spawn from 'cross-spawn';

describe('nx-python migrate-shared-venv generator', () => {
  let appTree: Tree;

  beforeEach(() => {
    vi.resetAllMocks();

    appTree = createTreeWithEmptyWorkspace({
      layout: 'apps-libs',
    });
    vi.mocked(spawn.sync).mockImplementation((command) => {
      if (command === 'python') {
        return {
          status: 0,
          output: [''],
          pid: 0,
          signal: null,
          stderr: null,
          stdout: Buffer.from('Python 3.8.11'),
        };
      }

      return {
        status: 0,
        output: [''],
        pid: 0,
        signal: null,
        stderr: null,
        stdout: null,
      };
    });
  });

  describe('poetry', () => {
    let checkPoetryExecutableMock: MockInstance;

    beforeEach(() => {
      checkPoetryExecutableMock = vi.spyOn(
        poetryUtils,
        'checkPoetryExecutable',
      );
      vi.spyOn(poetryUtils, 'getPoetryVersion').mockResolvedValue('1.8.2');
      checkPoetryExecutableMock.mockResolvedValue(undefined);
    });

    it('should throw an exception when the poetry is not installed', async () => {
      checkPoetryExecutableMock.mockRejectedValue(
        new Error('poetry not found'),
      );

      await expect(
        generator(appTree, {
          moveDevDependencies: true,
          pyenvPythonVersion: '3.8.11',
          pyprojectPythonDependency: '>=3.8,<3.10',
          autoActivate: false,
          packageManager: 'poetry',
        }),
      ).rejects.toThrow('poetry not found');

      expect(checkPoetryExecutableMock).toHaveBeenCalled();
    });

    it('should migrate an isolate venv to shared venv', async () => {
      await poetryProjectGenerator(appTree, {
        name: 'proj1',
        type: 'application',
        publishable: true,
        customSource: false,
        addDevDependencies: true,
        moduleName: 'proj1',
        packageName: 'proj1',
        buildLockedVersions: true,
        buildBundleLocalDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        toxEnvlist: 'py38',
      });

      const task = await generator(appTree, {
        moveDevDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        autoActivate: false,
        packageManager: 'poetry',
      });
      await task();

      expect(checkPoetryExecutableMock).toHaveBeenCalled();
      expect(
        appTree.read('apps/proj1/pyproject.toml', 'utf-8'),
      ).toMatchSnapshot();
      expect(appTree.read('pyproject.toml', 'utf-8')).toMatchSnapshot();
      expect(appTree.read('.python-version', 'utf-8')).toMatchSnapshot();
      expect(spawn.sync).toHaveBeenNthCalledWith(
        1,
        'poetry',
        ['lock', '--no-update'],
        { cwd: 'apps/proj1', shell: false, stdio: 'inherit' },
      );
      expect(spawn.sync).toHaveBeenNthCalledWith(
        2,
        'poetry',
        ['install', '-v'],
        {
          shell: false,
          stdio: 'inherit',
        },
      );
    });

    it('should migrate an isolate venv to shared venv (poetry 2.0.0)', async () => {
      vi.spyOn(poetryUtils, 'getPoetryVersion').mockResolvedValue('2.0.0');
      await poetryProjectGenerator(appTree, {
        name: 'proj1',
        type: 'application',
        publishable: true,
        customSource: false,
        addDevDependencies: true,
        moduleName: 'proj1',
        packageName: 'proj1',
        buildLockedVersions: true,
        buildBundleLocalDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        toxEnvlist: 'py38',
      });

      const task = await generator(appTree, {
        moveDevDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        autoActivate: false,
        packageManager: 'poetry',
      });
      await task();

      expect(checkPoetryExecutableMock).toHaveBeenCalled();
      expect(
        appTree.read('apps/proj1/pyproject.toml', 'utf-8'),
      ).toMatchSnapshot();
      expect(appTree.read('pyproject.toml', 'utf-8')).toMatchSnapshot();
      expect(appTree.read('.python-version', 'utf-8')).toMatchSnapshot();
      expect(spawn.sync).toHaveBeenNthCalledWith(1, 'poetry', ['lock'], {
        cwd: 'apps/proj1',
        shell: false,
        stdio: 'inherit',
      });
      expect(spawn.sync).toHaveBeenNthCalledWith(
        2,
        'poetry',
        ['install', '-v'],
        {
          shell: false,
          stdio: 'inherit',
        },
      );
    });

    it('should migrate an isolate venv to shared venv project without dev dependencies', async () => {
      await poetryProjectGenerator(appTree, {
        name: 'proj1',
        type: 'application',
        publishable: true,
        customSource: false,
        addDevDependencies: false,
        moduleName: 'proj1',
        packageName: 'proj1',
        buildLockedVersions: true,
        buildBundleLocalDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        toxEnvlist: 'py38',
      });

      const task = await generator(appTree, {
        moveDevDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        autoActivate: false,
        packageManager: 'poetry',
      });
      await task();

      expect(checkPoetryExecutableMock).toHaveBeenCalled();
      expect(
        appTree.read('apps/proj1/pyproject.toml', 'utf-8'),
      ).toMatchSnapshot();
      expect(appTree.read('pyproject.toml', 'utf-8')).toMatchSnapshot();
      expect(appTree.read('.python-version', 'utf-8')).toMatchSnapshot();
      expect(spawn.sync).toHaveBeenNthCalledWith(
        1,
        'poetry',
        ['lock', '--no-update'],
        { cwd: 'apps/proj1', shell: false, stdio: 'inherit' },
      );
      expect(spawn.sync).toHaveBeenNthCalledWith(
        2,
        'poetry',
        ['install', '-v'],
        {
          shell: false,
          stdio: 'inherit',
        },
      );
    });

    it('should migrate an isolate venv to shared venv with auto activate enabled', async () => {
      await poetryProjectGenerator(appTree, {
        name: 'proj1',
        type: 'application',
        publishable: true,
        customSource: false,
        addDevDependencies: true,
        moduleName: 'proj1',
        packageName: 'proj1',
        buildLockedVersions: true,
        buildBundleLocalDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        toxEnvlist: 'py38',
      });

      const task = await generator(appTree, {
        moveDevDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        autoActivate: true,
        packageManager: 'poetry',
      });
      task();

      expect(checkPoetryExecutableMock).toHaveBeenCalled();
      expect(appTree.read('pyproject.toml', 'utf-8')).toMatchSnapshot();
    });
  });

  describe('uv', () => {
    let checkUvExecutableMock: MockInstance;

    beforeEach(() => {
      checkUvExecutableMock = vi.spyOn(uvUtils, 'checkUvExecutable');
      checkUvExecutableMock.mockResolvedValue(undefined);
    });

    it('should throw an exception when the uv is not installed', async () => {
      checkUvExecutableMock.mockRejectedValue(new Error('uv not found'));

      await expect(
        generator(appTree, {
          moveDevDependencies: true,
          pyenvPythonVersion: '3.8.11',
          pyprojectPythonDependency: '>=3.8,<3.10',
          autoActivate: false,
          packageManager: 'uv',
        }),
      ).rejects.toThrow('uv not found');

      expect(checkUvExecutableMock).toHaveBeenCalled();
    });

    it('should migrate an isolate venv to shared venv', async () => {
      await uvProjectGenerator(appTree, {
        name: 'proj1',
        projectType: 'application',
        pyprojectPythonDependency: '',
        publishable: false,
        buildLockedVersions: false,
        buildBundleLocalDependencies: false,
        linter: 'ruff',
        unitTestRunner: 'pytest',
        rootPyprojectDependencyGroup: 'main',
        unitTestHtmlReport: true,
        unitTestJUnitReport: true,
        codeCoverage: true,
        codeCoverageHtmlReport: true,
        codeCoverageXmlReport: true,
        projectNameAndRootFormat: 'derived',
      });

      const task = await generator(appTree, {
        moveDevDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        autoActivate: false,
        packageManager: 'uv',
      });
      await task();

      expect(checkUvExecutableMock).toHaveBeenCalled();
      expect(
        appTree.read('apps/proj1/pyproject.toml', 'utf-8'),
      ).toMatchSnapshot();
      expect(appTree.read('pyproject.toml', 'utf-8')).toMatchSnapshot();
      expect(appTree.read('.python-version', 'utf-8')).toMatchSnapshot();
      expect(spawn.sync).toHaveBeenNthCalledWith(1, 'python', ['--version'], {
        stdio: 'pipe',
      });
      expect(spawn.sync).toHaveBeenNthCalledWith(2, 'uv', ['sync'], {
        shell: false,
        stdio: 'inherit',
      });
    });

    it('should migrate an isolate venv to shared venv project without dev dependencies', async () => {
      await uvProjectGenerator(appTree, {
        name: 'proj1',
        projectType: 'application',
        pyprojectPythonDependency: '',
        publishable: false,
        buildLockedVersions: false,
        buildBundleLocalDependencies: false,
        linter: 'none',
        unitTestRunner: 'none',
        rootPyprojectDependencyGroup: 'main',
        unitTestHtmlReport: false,
        unitTestJUnitReport: false,
        codeCoverage: false,
        codeCoverageHtmlReport: false,
        codeCoverageXmlReport: false,
        projectNameAndRootFormat: 'derived',
      });

      const task = await generator(appTree, {
        moveDevDependencies: true,
        pyenvPythonVersion: '3.8.11',
        pyprojectPythonDependency: '>=3.8,<3.10',
        autoActivate: false,
        packageManager: 'uv',
      });
      await task();

      expect(checkUvExecutableMock).toHaveBeenCalled();
      expect(
        appTree.read('apps/proj1/pyproject.toml', 'utf-8'),
      ).toMatchSnapshot();
      expect(appTree.read('pyproject.toml', 'utf-8')).toMatchSnapshot();
      expect(appTree.read('.python-version', 'utf-8')).toMatchSnapshot();
      expect(spawn.sync).toHaveBeenNthCalledWith(1, 'python', ['--version'], {
        stdio: 'pipe',
      });
      expect(spawn.sync).toHaveBeenNthCalledWith(2, 'uv', ['sync'], {
        shell: false,
        stdio: 'inherit',
      });
    });
  });
});
