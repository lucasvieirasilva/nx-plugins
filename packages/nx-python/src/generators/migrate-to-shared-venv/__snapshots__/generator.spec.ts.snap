// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv (poetry 2.0.0) 1`] = `
"[tool.coverage.run]
branch = true
source = [ "proj1" ]

[tool.coverage.report]
exclude_lines = [ "if TYPE_CHECKING:" ]
show_missing = true

[tool.pytest.ini_options]
addopts = "--cov --cov-fail-under=100 --cov-report html:'../../coverage/apps/proj1/html' --cov-report xml:'../../coverage/apps/proj1/coverage.xml' --junitxml='../../reports/apps/proj1/unittests/junit.xml' --html='../../reports/apps/proj1/unittests/html/index.html'"

[tool.poetry]
name = "proj1"
version = "1.0.0"
description = ""
authors = [ ]
license = "Proprietary"
readme = "README.md"
group = { }

  [[tool.poetry.packages]]
  include = "proj1"

  [tool.poetry.dependencies]
  python = ">=3.8,<3.10"

[build-system]
requires = [ "poetry-core==1.1.0" ]
build-backend = "poetry.core.masonry.api"
"
`;

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv (poetry 2.0.0) 2`] = `
"[tool.poetry]
name = "@nxlv/nx-plugins"
version = "1.0.0"
description = ""
authors = [ ]
license = "Proprietary"
readme = "README.md"
package-mode = false

  [tool.poetry.dependencies]
  python = ">=3.8,<3.10"

    [tool.poetry.dependencies.proj1]
    path = "apps/proj1"
    develop = true

[tool.poetry.group.dev.dependencies]
flake8 = "4.0.1"
flake8-isort = "4.1.1"
flake8-print = "5.0.0"
flake8-pytest-style = "1.6.0"
flake8-docstrings = "1.6.0"
flake8-type-checking = "2.0.6"
autopep8 = "1.5.7"
pytest = "7.1.2"
pytest-env = "0.6.2"
pytest-cov = "3.0.0"
pytest-html = "3.1.1"
pytest-sugar = "0.9.5"
tomli = "1.2.2"
tox = "3.23.1"

[build-system]
requires = [ "poetry-core==1.1.0" ]
build-backend = "poetry.core.masonry.api"
"
`;

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv (poetry 2.0.0) 3`] = `
"3.8.11
"
`;

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv 1`] = `
"[tool.coverage.run]
branch = true
source = [ "proj1" ]

[tool.coverage.report]
exclude_lines = [ "if TYPE_CHECKING:" ]
show_missing = true

[tool.pytest.ini_options]
addopts = "--cov --cov-fail-under=100 --cov-report html:'../../coverage/apps/proj1/html' --cov-report xml:'../../coverage/apps/proj1/coverage.xml' --junitxml='../../reports/apps/proj1/unittests/junit.xml' --html='../../reports/apps/proj1/unittests/html/index.html'"

[tool.poetry]
name = "proj1"
version = "1.0.0"
description = ""
authors = [ ]
license = "Proprietary"
readme = "README.md"
group = { }

  [[tool.poetry.packages]]
  include = "proj1"

  [tool.poetry.dependencies]
  python = ">=3.8,<3.10"

[build-system]
requires = [ "poetry-core==1.1.0" ]
build-backend = "poetry.core.masonry.api"
"
`;

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv 2`] = `
"[tool.poetry]
name = "@nxlv/nx-plugins"
version = "1.0.0"
description = ""
authors = [ ]
license = "Proprietary"
readme = "README.md"
package-mode = false

  [tool.poetry.dependencies]
  python = ">=3.8,<3.10"

    [tool.poetry.dependencies.proj1]
    path = "apps/proj1"
    develop = true

[tool.poetry.group.dev.dependencies]
flake8 = "4.0.1"
flake8-isort = "4.1.1"
flake8-print = "5.0.0"
flake8-pytest-style = "1.6.0"
flake8-docstrings = "1.6.0"
flake8-type-checking = "2.0.6"
autopep8 = "1.5.7"
pytest = "7.1.2"
pytest-env = "0.6.2"
pytest-cov = "3.0.0"
pytest-html = "3.1.1"
pytest-sugar = "0.9.5"
tomli = "1.2.2"
tox = "3.23.1"

[build-system]
requires = [ "poetry-core==1.1.0" ]
build-backend = "poetry.core.masonry.api"
"
`;

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv 3`] = `
"3.8.11
"
`;

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv project without dev dependencies 1`] = `
"[tool.coverage.run]
branch = true
source = [ "proj1" ]

[tool.coverage.report]
exclude_lines = [ "if TYPE_CHECKING:" ]
show_missing = true

[tool.pytest.ini_options]
addopts = "--cov --cov-fail-under=100 --cov-report html:'../../coverage/apps/proj1/html' --cov-report xml:'../../coverage/apps/proj1/coverage.xml' --junitxml='../../reports/apps/proj1/unittests/junit.xml' --html='../../reports/apps/proj1/unittests/html/index.html'"

[tool.poetry]
name = "proj1"
version = "1.0.0"
description = ""
authors = [ ]
license = "Proprietary"
readme = "README.md"

  [[tool.poetry.packages]]
  include = "proj1"

  [tool.poetry.dependencies]
  python = ">=3.8,<3.10"

[build-system]
requires = [ "poetry-core==1.1.0" ]
build-backend = "poetry.core.masonry.api"
"
`;

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv project without dev dependencies 2`] = `
"[tool.poetry]
name = "@nxlv/nx-plugins"
version = "1.0.0"
description = ""
authors = [ ]
license = "Proprietary"
readme = "README.md"
package-mode = false

  [tool.poetry.dependencies]
  python = ">=3.8,<3.10"

    [tool.poetry.dependencies.proj1]
    path = "apps/proj1"
    develop = true

[build-system]
requires = [ "poetry-core==1.1.0" ]
build-backend = "poetry.core.masonry.api"
"
`;

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv project without dev dependencies 3`] = `
"3.8.11
"
`;

exports[`nx-python migrate-shared-venv generator > poetry > should migrate an isolate venv to shared venv with auto activate enabled 1`] = `
"[tool.nx]
autoActivate = true

[tool.poetry]
name = "@nxlv/nx-plugins"
version = "1.0.0"
description = ""
authors = [ ]
license = "Proprietary"
readme = "README.md"
package-mode = false

  [tool.poetry.dependencies]
  python = ">=3.8,<3.10"

    [tool.poetry.dependencies.proj1]
    path = "apps/proj1"
    develop = true

[tool.poetry.group.dev.dependencies]
flake8 = "4.0.1"
flake8-isort = "4.1.1"
flake8-print = "5.0.0"
flake8-pytest-style = "1.6.0"
flake8-docstrings = "1.6.0"
flake8-type-checking = "2.0.6"
autopep8 = "1.5.7"
pytest = "7.1.2"
pytest-env = "0.6.2"
pytest-cov = "3.0.0"
pytest-html = "3.1.1"
pytest-sugar = "0.9.5"
tomli = "1.2.2"
tox = "3.23.1"

[build-system]
requires = [ "poetry-core==1.1.0" ]
build-backend = "poetry.core.masonry.api"
"
`;

exports[`nx-python migrate-shared-venv generator > uv > should migrate an isolate venv to shared venv 1`] = `
"[tool.coverage.run]
branch = true
source = [ "proj1" ]

[tool.coverage.report]
exclude_lines = [ "if TYPE_CHECKING:" ]
show_missing = true

[tool.pytest.ini_options]
addopts = """
 --cov
 --cov-report html:'../../coverage/apps/proj1/html'
 --cov-report xml:'../../coverage/apps/proj1/coverage.xml'
 --html='../../reports/apps/proj1/unittests/html/index.html'
 --junitxml='../../reports/apps/proj1/unittests/junit.xml'
"""

[tool.hatch.build.targets.wheel]
packages = [ "proj1" ]

[tool.hatch.metadata]
allow-direct-references = true

[tool.ruff]
exclude = [ ".ruff_cache", ".svn", ".tox", ".venv", "dist" ]
line-length = 88
indent-width = 4

  [tool.ruff.lint]
  select = [ "E", "F", "UP", "B", "SIM", "I" ]
  ignore = [ ]
  fixable = [ "ALL" ]
  unfixable = [ ]

[project]
name = "proj1"
version = "1.0.0"
description = "Automatically generated by Nx."
requires-python = ">=3.9,<4"
readme = "README.md"
dependencies = [ ]

[build-system]
requires = [ "hatchling" ]
build-backend = "hatchling.build"
"
`;

exports[`nx-python migrate-shared-venv generator > uv > should migrate an isolate venv to shared venv 2`] = `
"[project]
name = "nx-workspace"
version = "1.0.0"
dependencies = [ "proj1" ]

[tool.uv.workspace]
members = [ "apps/proj1" ]

[tool.uv.sources.proj1]
workspace = true

[dependency-groups]
dev = [
  "autopep8>=2.3.1",
  "ruff>=0.8.2",
  "pytest>=8.3.4",
  "pytest-sugar>=1.0.0",
  "pytest-cov>=6.0.0",
  "pytest-html>=4.1.1"
]
"
`;

exports[`nx-python migrate-shared-venv generator > uv > should migrate an isolate venv to shared venv 3`] = `
"3.8.11
"
`;

exports[`nx-python migrate-shared-venv generator > uv > should migrate an isolate venv to shared venv project without dev dependencies 1`] = `
"[project]
name = "proj1"
version = "1.0.0"
description = "Automatically generated by Nx."
requires-python = ">=3.9,<4"
readme = "README.md"
dependencies = [ ]

[tool.hatch.build.targets.wheel]
packages = [ "proj1" ]

[tool.hatch.metadata]
allow-direct-references = true

[build-system]
requires = [ "hatchling" ]
build-backend = "hatchling.build"
"
`;

exports[`nx-python migrate-shared-venv generator > uv > should migrate an isolate venv to shared venv project without dev dependencies 2`] = `
"[project]
name = "nx-workspace"
version = "1.0.0"
dependencies = [ "proj1" ]

[tool.uv.workspace]
members = [ "apps/proj1" ]

[tool.uv.sources.proj1]
workspace = true

[dependency-groups]
dev = [ "autopep8>=2.3.1" ]
"
`;

exports[`nx-python migrate-shared-venv generator > uv > should migrate an isolate venv to shared venv project without dev dependencies 3`] = `
"3.8.11
"
`;
